src/test/e2e/28_wait_test.go

```diff
@@ -0,0 +1,61 @@
+package test
+
+import (
+	// "context"
+	"fmt"
+	"time"
+
+	// "os/exec"
+	"testing"
+
+	// "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type zarfCommandResult struct {
+	stdOut string
+	stdErr string
+	err    error
+}
+
+func zarfCommandWStruct(e2e ZarfE2ETest, path string) (result zarfCommandResult) {
+	result.stdOut, result.stdErr, result.err = e2e.execZarfCommand("package", "deploy", path, "--confirm")
+	return result
+}
+
+func TestWait(t *testing.T) {
+	t.Log("E2E: Helm Wait")
+	e2e.setup(t)
+	defer e2e.teardown(t)
+
+	path := fmt.Sprintf("build/zarf-package-test-helm-wait-%s.tar.zst", e2e.arch)
+
+	zarfChannel := make(chan zarfCommandResult, 1)
```

Comment:
@jeff-mccoy @Racer159 Was there a better way to go about setting a timeout for a function, I guess its not too bad now that I wrote it but figuring this out took a really long time lol

Reply:
I think this looks fine from an implementation perspective (especially for what you are trying to do).  The only other big place in tests we have that watches for a timeout is here: https://github.com/defenseunicorns/zarf/blob/2a6bdbd75a4d3d118f18ae20b8c3aed2e355a8cf/src/test/e2e/22_git_and_flux_test.go#L103 but that doesn't always work smoothly (and is checking something after the Zarf command has finished and Flux takes over)